File: C++/프로그래머스/2/64065. 튜플/튜플.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

vector<int> solution(string s) {
    unordered_map<int, int> nmap;
    string tmp = ""; 
    for (auto& c : s) {
        if(isdigit(c)) {
            tmp += c;
        } else if((c == ',' || c == '}') && !tmp.empty()) {
            nmap.insert({stoi(tmp), 0});
            nmap[stoi(tmp)]++;
            tmp = "";
        }
    }
    
    vector<pair<int, int>> vec(nmap.begin(), nmap.end());
    sort(vec.begin(), vec.end(), [&](pair<int, int> a, pair<int, int> b) -> bool {
        return a.second > b.second;
    });
    
    
    vector<int> answer;
    for(auto& v : vec) {
        answer.push_back(v.first);
    }
    return answer;
}

File: Python3/프로그래머스/2/64065. 튜플/튜플.py
def solution(s):
    alist = s[2:-2].split("},{")
    alist.sort(key=lambda x: len(x))
    
    answer = []
    for a in alist:
        a = a.split(",")
        for x in a:
            x = int(x)
            if x not in answer:
                answer.append(x)
    
    return answer